### PR TITLE
Add simple store as dependencies

### PR DESCRIPTION
**Pull request by cychiuae**
@louischan-oursky

### CODE REVIEW

File: pkg/api/oauthrelyingparty/provider.go [hunk 1]
```diff
@@ -146,11 +146,17 @@ type Clock interface {
 	NowUTC() time.Time
 }
 
+type SimpleStore interface {
+	SetWithTTL(key string, value string, ttl time.Duration) error
+	GetDel(key string) (string, error)
```
Comment (louischan-oursky) [inline pkg/api/oauthrelyingparty/provider.go:151]: I suppose the implementation of this store will prefix the key with `app:{app_id}:` right? If that is the case, then it looks good to me.

Comment (cychiuae) [reply pkg/api/oauthrelyingparty/provider.go:151]: It would be `app:{app_id}:oauth:{provider_type}:`

Comment (louischan-oursky) [reply pkg/api/oauthrelyingparty/provider.go:151]: It would be better if it is `app:{app_id}:oauth:{provider_type}:{provider_alias}:`

Comment (cychiuae) [reply pkg/api/oauthrelyingparty/provider.go:151]: Sure

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**